src/vs/workbench/contrib/remote/browser/remoteExplorer.ts

```diff
@@ -524,14 +532,13 @@ class ProcAutomaticPortForwarding extends Disposable {
 			this.portsFeatures.dispose();
 		}
 
-		if (!this.remoteExplorerService.tunnelModel.environmentTunnelsSet) {
-			await new Promise<void>(resolve => this.remoteExplorerService.tunnelModel.onEnvironmentTunnelsSet(() => resolve()));
-		}
-
 		// Capture list of starting candidates so we don't auto forward them later.
 		await this.setInitialCandidates();
 
-		this.candidateListener = this._register(this.remoteExplorerService.tunnelModel.onCandidatesChanged(this.handleCandidateUpdate, this));
+		// Need to check the setting again, since it may have changed while we waited for the initial candidates to be set.
+		if (this.configurationService.getValue(PORT_AUTO_FORWARD_SETTING)) {
```

Comment:
We do check `PORT_AUTO_FORWARD_SETTING` before calling `startCandidateListner`; however, the setting value can change while we're awaiting the `setInitialCandidates()` just above. This can cause auto port forwarding to start even though the setting was `true` before entering `startCandidateListener`.